### PR TITLE
Fix false-positive when merging unions with `+`

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/phpstan/phpstan/issues/5584

### CODE REVIEW

File: src/Analyser/MutatingScope.php [hunk 1]
```diff
@@ -1321,58 +1373,6 @@ private function resolveType(Expr $node): Type
 				}
 			}
 
-			if ($node instanceof Expr\AssignOp\Plus || $node instanceof Expr\BinaryOp\Plus) {
```
Comment (staabm) [inline src/Analyser/MutatingScope.php:1324]: just moved this block before the integer-range math

File: tests/PHPStan/Analyser/data/math.php [hunk 2]
```diff
@@ -109,4 +109,20 @@ public function doSit(int $i, int $j): void
 		assertType('int', $i - $j);
 	}
 
+	public function unionSum(): void
```
Comment (ondrejmirtes) [inline tests/PHPStan/Analyser/data/math.php:112]: When solving a specific issue, please always create a new file called `bug-xxxx.php`, thanks. The namespace in the file should also match.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: tests/PHPStan/Analyser/data/math.php (hunk 2 above)
Comment (staabm) [reply tests/PHPStan/Analyser/data/math.php]: done

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**